### PR TITLE
Feature/#21 dual axes domain

### PR DESCRIPTION
**Pull request by adarshlilha**
Add a method allowing user to add config for axes separately.

**Usage**
```javascript
.config({
            axes: {
                x: function (rowIndex, columnIndex, context) {
                    if (context.facetFields[0][0] === 'Origin' && context.facetFields[1][0] === 'USA') {
                        return {
                            tickFormat: (d) => `${d}K`,
                        }
                    }
                    // Apply to all x-axes
                    return {
                        // name: 'hello world',
                        tickFormat: (d) => `${d / 1000}K`,
                    };
                }
            }
        });
    
        setTimeout(() => {
            canvas
                .config({
                    axes: {
                        x: function (rowIndex, columnIndex, context) {
                            // Update only one axis
                            // Using any of the below if statements
                            // if (rowIndex === 1 && columnIndex === 0) {} OR
                            if (context.axisFields[0] === 'Displacement' && context.facetFields[1][0] === 'USA') {
                                return {
                                    domain: [0, 14000],
                                    name: 'New Name'
                                };
                            }
                            return null;
                        },
                    }
                });
        }, 2000);
```
x-axes targeted in `setTimeout` fn is updated while all others do not change

### CODE REVIEW

File: packages/visual-group/src/encoder/encoder-helper.js [hunk 1]
```diff
@@ -128,7 +140,7 @@ export const generateAxisFromMap = (axisType, fieldInfo, axesCreators, axesInfo)
             axis = map.get(axisKey);
             axis._rotationLock = false;
             axis.config(axisConfig);
-            axisConfig.domain ? axis.domain(axisConfig.domain) : axis.resetDomain();
+            axisConfig && axisConfig.domain ? axis.domain(axisConfig.domain) : axis.resetDomain();
```
[the remark this answers is not in the document]
Comment (adarshlilha) [reply packages/visual-group/src/encoder/encoder-helper.js]: removed the call

File: packages/visual-group/src/encoder/encoder-helper.js [hunk 2]
```diff
@@ -13,11 +13,23 @@ import { CATEGORICAL, TEMPORAL, BAR, LINE, POINT, BOTH, Y } from '../enums/const
...
+    const { rowIndex, columnIndex } = indices;
+    const rawUserAxisConfig = config.axes ? (config.axes[axisType] || {}) : {};
+
+    // Change config object to a function if not already one
+    const userAxisConfigFn = typeof rawUserAxisConfig !== 'function' ?
+        () => rawUserAxisConfig : rawUserAxisConfig;
+    const userAxisConfig = userAxisConfigFn(field._oneVar, rowIndex, columnIndex);
+
+    // If current config does not specifes config for an axis, retain old config
+    if (!userAxisConfig) {
```
[the remark this answers is not in the document]
Comment (adarshlilha) [reply packages/visual-group/src/encoder/encoder-helper.js]: check if needed here to retain the old axis config

File: packages/muze-axis/src/cartesian-axis/simple-axis.js [hunk 3]
```diff
@@ -299,9 +299,8 @@ export default class SimpleAxis {
     getLogicalSpace () {
         if (!this.logicalSpace()) {
```
Comment (ranajitbanerjee) [inline packages/muze-axis/src/cartesian-axis/simple-axis.js]: `this.logicalSpace()` is called two times here

Comment (adarshlilha) [reply packages/muze-axis/src/cartesian-axis/simple-axis.js]: `logicalSpace` setter is called if it is not already set, else it is called as a getter whose value is returned

**Review by ranajitbanerjee — APPROVED** [review]
Working fine

**Pull request closed without merging**